src/Language/PureScript/Bridge/Printer.hs

```diff
@@ -83,9 +83,9 @@ sumTypeToText st@(SumType t cs) = T.unlines $
     genericConstrainsInner = T.intercalate ", " $ map genericInstance sumTypeParameters
     stpLength = length sumTypeParameters
     bracketWrap x = "(" <> x <> ")"
-    sumTypeParameters = filter isTypeParameter $ getUsedTypes st
-    -- TODO: I find this stupid. Would much rather like some other way to make this check?
-    isTypeParameter tp = T.toLower (_typeName tp) == _typeName tp
+    sumTypeParameters = filter isTypeParam . Set.toList $ getUsedTypes st
+    -- NOTE: only this captures monadic type parameters. See test/Spec.hs
+    isTypeParam typ = T.toLower (_typeName typ) == _typeName typ
```

Comment:
Its done in https://github.com/eskimor/purescript-bridge/pull/12/commits/0f3ca880c2949908b776fabf2bdbc11628878a02